plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeConnectorSmokeTest.java

```diff
@@ -1959,6 +1959,78 @@ public void testHistoryTable()
         }
     }
 
+    @Test
+    public void testHistoryTableWithDeletedTransactionLog()
+            throws Exception
+    {
+        try (TestTable table = new TestTable(
+                getQueryRunner()::execute,
+                "test_history_table_with_deleted_transaction_log",
```

Comment:
You can also use `checkpoint_interval` set to a lower value to avoid doing too many inserts.